project_template/spec/spec_helper.coffee

```diff
@@ -1,3 +1,29 @@
 Backbone = require('backbone')
 # Backbone.$ is not automatically set when backbone is loaded as CommonJS module
 $ = Backbone.$ = require('jquery')
+
+# Allow to use memo-isation in tests
+#
+# describe 'some feature', ->
+#
+#   data = memo().is -> { foo: 'bar' }
+#
+#   it 'shows bar', ->
+#     expect(data().foo).to.eql 'bar'
+#
+#   describe 'when undefined', ->
+#     data.is -> { foo: undefined }
+#
+#     it 'shows undefined', ->
+#       expect(data().foo).to.be.undefined
+#
+window.memo = require('memo-is')
+
+When = require('when')
+window.wait = (delay = 1) ->
```

Comment:
All Maji projects I've done so far have tests that kick off async processes in views.
So in the test you interact with the view through their DOM. and have no means to wait for kicked off async processes (be it a promise to a native service, a backend, or an animation)

So in these tests you have to wait for arbitrary amounts anyway. Then the question becomes:

would I like my test to look like this:

```coffeescript
it 'slides open a menu', (done) ->
  @view.$('a').click()
  setTimeout(
    =>
      expect(something).to.have.happened
      done()
    arbitraryWait
  )
```

Or like this?

```coffeescript
it 'slides open a menu', ->
  @view.$('a').click()
  wait(arbitraryWait).then ->
    expect(something).to.have.happened
```

I choose the latter one. When we don't provide these means, these tests either: will not be written at all (too hard?) or will be unreadable.